webui/react/src/hooks/useSettings.ts

```diff
@@ -291,38 +294,52 @@ const useSettings = <T>(config: SettingsConfig<T>): UseSettingsReturn<T> => {
   );
 
   const updateSettings = useCallback(
-    async (updates: Settings, shouldPush = false) => {
-      if (!settings || shouldSkipUpdates) return;
+    (updates: Settings, shouldPushUpdate = false) => {
+      if (querySettings || shouldSkipUpdates) return; // prevent changing the settings state if getting it from querrySettings or if it shouldSkipUpdates
 
-      const newSettings = { ...settings, ...updates };
+      update(config.storagePath, (settings) => {
+        if (!settings) return updates;
 
-      if (isEqual(newSettings, settings)) return;
+        return { ...settings, ...updates };
+      });
 
-      update(config.storagePath, newSettings);
+      setShouldPush(shouldPushUpdate);
+    },
+    [config, update, shouldSkipUpdates, querySettings],
+  );
 
-      await updateDB(newSettings);
+  useEffect(() => {
+    if (!settings || querySettings) return;
 
-      if (
-        (Object.values(config.settings) as SettingsConfigProp<typeof config>[]).every(
-          (setting) => !!setting.skipUrlEncoding,
-        )
-      ) {
-        return;
+    setReturnedSettings((prevState) => {
+      if (isEqual(prevState, settings)) {
+        return prevState;
       }
 
-      const mappedSettings = settingsToQuery(config, newSettings);
-      const url = `?${mappedSettings}`;
+      updateDB(settings);
```

Comment:
i dont think its a good idea to have `updateDB(settings);` inside `setReturnedSettings`